optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -317,6 +329,45 @@ private void readMeetingDuration(Meeting meeting) {
                     .collect(toList());
         }
 
+        private TimeGrain extractTimeGrain(Meeting meeting, Map<LocalDateTime, TimeGrain> timeGrainMap) {
+            String dateString = nextStringCell().getStringCellValue();
+            String startTimeString = nextStringCell().getStringCellValue();
+            if (!dateString.isEmpty() || !startTimeString.isEmpty()) {
+                LocalDateTime dateTime;
+                try {
+                    dateTime = LocalDateTime.of(LocalDate.parse(dateString, DAY_FORMATTER),
+                            LocalTime.parse(startTimeString, TIME_FORMATTER));
+                } catch (DateTimeParseException e) {
+                    throw new IllegalStateException(currentPosition() + ": The meeting with id (" + meeting.getId()
+                            + ") has a timeGrain date (" + dateString + ") and startTime (" + startTimeString
+                            + ") that doesn't parse as a date or time.", e);
+                }
+
+                TimeGrain timeGrain = timeGrainMap.get(dateTime);
+                if (timeGrain == null) {
+                    throw new IllegalStateException(currentPosition() + ": The meeting with id (" + meeting.getId()
+                            + ") has a timeGrain date (" + dateString + ") and startTime (" + startTimeString
+                            + ") that doesn't exist in the other sheet (Day).");
+                }
+                return timeGrain;
+            }
+            return null;
```

Comment:
soft suggestion: returning null on one place might result in NPE in other place. `Optional.empty()` could be used in such situations.

Reply:
if the field is empty, the timegrain of the MeetingAssignment should be null